tracer/src/Datadog.Trace/Ci/GitCommandHelper.cs

```diff
@@ -188,6 +189,222 @@ static List<FileCoverageInfo> ParseGitDiff(string diffOutput)
+
+            if (verifyBranchOutput is not { ExitCode: 0 })
+            {
+                Log.Warning("GitCommandHelper: Branch '{Branch}' does not exist", targetBranch);
+                return null;
+            }
+
+            // 2. Check if the target is already a main-like branch
+            string shortTargetName = targetBranch!;
+            if (shortTargetName.StartsWith($"{remoteName}/"))
+            {
+                shortTargetName = shortTargetName.Substring(remoteName.Length + 1);
+            }
+
+            if (Regex.IsMatch(shortTargetName, branchFilterPattern))
+            {
+                Log.Debug("GitCommandHelper: Branch '{Branch}' already matches branch filter → no parent needed", targetBranch);
+                return null;
+            }
+
+            // 3. Detect default branch
+            string? defaultBranch = null;
+
+            // Try to get the symbolic reference for origin/HEAD
+            var symbolicRefOutput = RunGitCommand(
+                workingDirectory,
+                $"symbolic-ref --quiet --short refs/remotes/{remoteName}/HEAD",
+                MetricTags.CIVisibilityCommands.GetSymbolicRef);
+
+            if (symbolicRefOutput is { ExitCode: 0 } && !string.IsNullOrWhiteSpace(symbolicRefOutput.Output))
+            {
+                var symbolicRef = symbolicRefOutput.Output.Trim();
+                string prefix = $"{remoteName}/";
+                if (symbolicRef.StartsWith(prefix))
+                {
+                    defaultBranch = symbolicRef.Substring(prefix.Length);
+                }
+                else
+                {
+                    defaultBranch = symbolicRef;
+                }
+            }
+            else
+            {
+                // Fallback to main or master
+                foreach (var fallback in new[] { "main", "master" })
+                {
+                    var fallbackOutput = RunGitCommand(
+                        workingDirectory,
+                        $"show-ref --verify --quiet refs/remotes/{remoteName}/{fallback}",
+                        MetricTags.CIVisibilityCommands.ShowRef);
+
+                    if (fallbackOutput is { ExitCode: 0 })
+                    {
+                        defaultBranch = fallback;
+                        break;
+                    }
+                }
+            }
+
+            // 4. Build candidate list
+            var branchesOutput = RunGitCommand(
+                workingDirectory,
+                $"for-each-ref --format='%(refname:short)' refs/heads refs/remotes/{remoteName}",
+                MetricTags.CIVisibilityCommands.BuildCandidateList);
+
+            if (branchesOutput is not { ExitCode: 0 } || string.IsNullOrWhiteSpace(branchesOutput?.Output))
+            {
+                Log.Warning("GitCommandHelper: Failed to get branch list");
+                return null;
+            }
+
+            var regex = new Regex(branchFilterPattern);
```

Comment:
Although I know this is in the original script, and is done for performance reasons to reduce the possible base branches to the ones following the pattern. I think we might need or to improve the pattern to allow more base branches or find another way to optimize it.

Reply:
Yeah.. I do not like it either at all.. For now I've moved the regex to a compiled constant instead of a parameter like in the script since I doubt that functionality will be ever used and it improves the performance a bit